### ISSUE TITLE
Switching view mode resets shop

### ISSUE THREAD

**Issue by GoogleCodeExporter** [alistairreilly/andors-trail#94]
```
What steps will reproduce the problem?
1. Sell something to a shop/trader
2. Switch from portrait to landscape mode (may apply vice versa too) 
3. The sold items are now gone. Shop has been reset (may apply to buy too)

What is the expected output? What do you see instead?
See the item(s) I sold still among on the shop list.

What version of the product are you using? On what operating system?
v0.6.7b2

Please provide any additional information below.
A little off-topic:
There seem to be number of issues about view mode. Is this the code or an 
Android thing ?
```

Original issue reported on code.google.com by `kims...@gmail.com` on 19 Dec 2010 at 11:42

**Comment by GoogleCodeExporter** [comment]
```
I don't think it's so serious. If you leave and re-enter the shop (i.e. talk to 
shopkeeper again), shop is re-stashed. I don't know if it is intentional though.
```

Original comment by `surrano` on 20 Dec 2010 at 8:59

**Comment by GoogleCodeExporter** [comment]
```
Yes, shop contents are not persisted in the savegame, they are recreated every 
time the ShopActivity is created. We currently don't have any good indicator as 
to when the shops should restock, and now we don't have to save the contents. 
Might change in the future though - but at the moment, shops are restocked when 
the list is shown.

re side note: This is an android thing. Activities (the visible part of the 
program) are restarted when the orientation changes. This is usually not an 
issue, and quite a smart move on the Android OS design since layouts might 
change and whatnot.
```

Original comment by `oskar.wi...@gmail.com` on 20 Dec 2010 at 9:28

**Comment by GoogleCodeExporter** [comment]
```
Might be an issue if you accidentally sell the wrong thing and then switch 
modes, no way to buy it back.
```

Original comment by `sdeva...@gmail.com` on 10 Feb 2011 at 1:33
- Added labels: **Priority-Low**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
Either we should never restrict the amount of an item available in a shop.
Or the amount should not refill. (or only over time)
```

Original comment by `SamuelPl...@gmail.com` on 9 Jul 2011 at 10:09

**Comment by GoogleCodeExporter** [comment]
```
Issue 211 has been merged into this issue.
```

Original comment by `sdeva...@gmail.com` on 9 Jul 2011 at 7:01

**Comment by GoogleCodeExporter** [comment]
```
Persistent shop inventories released in v0.6.12.
```

Original comment by `oskar.wi...@gmail.com` on 28 Oct 2012 at 7:37
- Changed state: **Fixed**
- Added labels: **Milestone-0.6.12**
